merkle/compact/tree.go

```diff
@@ -26,7 +26,7 @@ import (
 	"github.com/google/trillian/merkle/hashers"
 )
 
-// NodeID identifies a node of a Merkle tree.
+// NodeID identifies a node of a log type Merkle tree.
 type NodeID struct {
 	Level uint
```

Comment:
While you're commenting, I would find it useful as an outsider to this code to have a 1-liner explaining where you are counting from with these values. Is Level 0 the root or leaf? Is Index counting from the left or right?

Reply:
Moved this type to `range.go` and reused an existing comment explaining the tree geometry.